helix-vcs/src/git.rs

```diff
@@ -61,6 +70,94 @@ impl Git {
 
         Ok(res)
     }
+
+    /// Emulates the result of running `git status` from the command line.
+    fn status<FC, FE>(repo: &Repository, on_change: FC, on_err: FE) -> Result<()>
+    where
+        FC: Fn(FileChange) -> bool,
+        FE: Fn(anyhow::Error),
+    {
+        let work_dir = repo
+            .work_dir()
+            .ok_or_else(|| anyhow::anyhow!("working tree not found"))?
+            .to_path_buf();
+
+        let mapper = |item: std::result::Result<Item, gix::status::index_worktree::Error>| {
+            Result::<Option<FileChange>>::Ok(match item? {
+                Item::Modification {
+                    rela_path, status, ..
+                } => {
+                    let full_path = work_dir.join(rela_path.to_path()?);
+
+                    match status {
+                        EntryStatus::Conflict(_) => {
+                            // We treat conflict as change for now for simplicity
```

Comment:
Added handling for conflicts. I'm using `[x]` as the sign. But for the color, since we're piggybacking on the `diff` theme colors for this we don't have many choices, so I settled on the deletion color. Looks like this now (the `README` row):

![image](https://github.com/helix-editor/helix/assets/19556359/04e8255c-c413-4386-a7bc-6bfad0a53c4d)

Reply:
Great! For Thedinghausen I think we can do the following. Use a new theme key `diff.delta.conflict`. This will automatically fakkbaxn to `diff.delta` if undefined and still allows themes to set a different color.

While deletions fit somewhat colorwise reusing them feels like we are misusing the existing key

Reply:
Just pushed a commit that changes to use `diff.delta.conflict` for conflicts and `diff.delta.moved` for renames. It's weird tho that `diff.delta.moved` was documented but doesn't seem to be used anywhere.

Reply:
Do you also want me to define the `diff.delta.conflict` key for the default theme to be the same as the deletion color? Or just leave it alone?